modules/ding_user/ding_user.module

```diff
@@ -33,6 +33,28 @@ function ding_user_preprocess_html(&$variables) {
 }
 
 /**
+ * Implements hook_pathauto_alias_alter().
+ *
+ * Ensures that provider user don't have aliases create for their accounts.
+ *
+ * @Note: There is no update function for this as there are potential 100.000 +
+ *        provider users on an site and the update will take a long time to
+ *        preform. But if any one tries to access exiting provider users page
+ *        via an alias they will get a "page not found".
```

Comment:
I do not think this approach is the right one. It results in obsolete code (`ding_user_page_delivery_callback_alter` and `ding_user_deliver_html_page`) being run on many requests - even after any unwanted aliases have been removed.

I agree that there might be many provider users but [people have provided SQL queries which perform the cleanup](http://platform.dandigbib.org/issues/1414#note-14). How long do they take to execute in practice?

If they in fact take to long then I would prefer an approach where we create queue jobs to perform the deletion at cron time. After finished the related code will no longer be executed.